libs/io/tests/test_h5.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# coding: utf-8
+import os
+import shutil
+from pathlib import Path
+from typing import Optional
+
+import h5py
+import numpy as np
+import pytest
+
+from bbhnet.io import h5
+
+
+@pytest.fixture(scope="function")
+def write_dir():
+    write_dir = "tmp"
+    os.makedirs(write_dir, exist_ok=True)
+    yield Path(write_dir)
+    shutil.rmtree(write_dir)
+
+
+@pytest.fixture(scope="function")
+def t():
+    t = [
+        1238163456,
+        12345677890,
+        12345677890,
+        12345677890,
+        12345677890,
+        1238167552,
+    ]
+    return t
+
+
+@pytest.fixture(scope="function")
+def y():
+    y = [23, 53.5, 234, 0, 5, 456.345]
+    return y
+
+
+@pytest.fixture(scope="function")
+def datasets():
+    datasets = {
+        "dataset1": np.arange(6),
```

Comment:
All the datasets should have the same length (see my comment below), which should match the length of `t` (`y` should match this length as well).

Reply:
Tests are included to check this